packages/atomic/src/components/product-listing/atomic-product-listing/atomic-product-listing.tsx

```diff
@@ -0,0 +1,292 @@
+      initialState: {numberOfResults: this.numberOfProducts},
+    });
+    this.productListing = buildProductListing(this.bindings.engine, {
+      options: {
+        url: this.bindings.store.getUrl(),
+        additionalFields: this.bindings.store.getFieldsToInclude(),
+      },
+    });
+    const resultTemplateProvider = new ResultTemplateProvider({
+      includeDefaultTemplate: true,
+      templateElements: Array.from(
+        this.host.querySelectorAll('atomic-product-listing-result-template')
+      ),
+      getResultTemplateRegistered: () => this.resultTemplateRegistered,
+      getTemplateHasError: () => this.templateHasError,
+      setResultTemplateRegistered: (value: boolean) => {
+        this.resultTemplateRegistered = value;
+      },
+      setTemplateHasError: (value: boolean) => {
+        this.templateHasError = value;
+      },
+      bindings: this.bindings,
+    });
+
+    this.resultListCommon = new ResultListCommon({
+      resultTemplateProvider,
+      getNumberOfPlaceholders: () => this.numberOfProducts,
+      host: this.host,
+      bindings: this.bindings,
+      getDensity: () => this.density,
+      getLayoutDisplay: () => 'grid',
+      getResultDisplay: () => this.display,
+      getImageSize: () => this.imageSize,
+      nextNewResultTarget: this.nextNewResultTarget,
+      loadingFlag: this.loadingFlag,
+      getResultListState: () => this.resultListCommonState,
+      getResultRenderingFunction: () => this.resultRenderingFunction,
+      renderResult: (props) => (
+        <atomic-product-listing-result
+          {...props}
+        ></atomic-product-listing-result>
+      ),
+      getInteractiveResult: (result: Result) =>
+        buildProductListingInteractiveResult(this.bindings.engine, {
+          options: {result},
+        }),
+    });
+  }
+
+  private get resultListCommonState(): ResultListCommonState<Result> {
+    return {
+      firstSearchExecuted: this.productListingState.responseId !== '',
+      isLoading: this.productListingState.isLoading,
+      hasError: this.productListingState.error !== null,
+      hasResults: this.productListingState.products.length !== 0,
+      results: this.productRecommendationsAsResuls,
+      searchResponseId: this.productListingState.responseId,
+    };
+  }
+
+  private updateOriginLevel2() {
+    if (this.label) {
+      const action = loadConfigurationActions(
+        this.bindings.engine
+      ).setOriginLevel2({
+        originLevel2: this.label,
+      });
+
+      this.bindings.engine.dispatch(action);
+    }
+  }
+
+  // private get subsetRecommendations() {
+  //   if (!this.numberOfRecommendationsPerPage) {
+  //     return this.productListingState.products;
+  //   }
+
+  //   return this.productListingState.products.slice(
+  //     this.currentIndex,
+  //     this.currentIndex + this.numberOfRecommendationsPerPage
+  //   );
+  // }
+
+  private get productRecommendationsAsResuls(): Result[] {
```

Comment:
Since the ProductListingEngine doesn't use the Result interface like the other engines I had to do this to use the common stuff. This is something I'd like to change in Headless in the future if possible. Would be easier if we were like everyone else.

Reply:
Why does it need to respect the `Result` interface coming from the classic search use case though ?

A product recommendation is not really equal to a standard search result 🤔

Reply:
I see... I guess we can merge it like that then, but an alternative would be to rework `ResultListCommon` to be more lax on the types needed/more specific on the properties it needs, versus requiring a "full" result with all the properties.